filters/ratelimit/ratelimit.go

```diff
@@ -457,6 +460,12 @@ func getStatusCodeArg(args []interface{}, index int) (int, error) {
 
 // Request checks ratelimit using filter settings and serves `429 Too Many Requests` response if limit is reached
 func (f *filter) Request(ctx filters.FilterContext) {
+	e, ok := ctx.StateBag()[FailClosedKey]
+	if ok {
+		b, ok := e.(bool)
+		f.settings.FailClosed = ok && b
+	}
```

Comment:
I am not sure modifying `f.settings` is thread safe here.
Maybe we can utilize `sync.Once` or even a route `PreProprocessor` as we only need to configure it once. I guess this is the downside of having separate filter for the configuration.

Reply:
done with PostProcessor